### PR TITLE
[RUN-5206] Add BrowserView Method Events

### PR DESCRIPTION
**Pull request by MichaelMCoates**
#### Description of Change

Adds the following emitted events for BrowserView Methods: 
`created`
`attached`
`shown`
`hidden`
`destroyed`

Tests:
https://testing-dashboard.openfin.co/#/app/sessions/api/completed/5d8124127ba87401eaf564a3

Associated core PR:
https://github.com/HadoukenIO/core/pull/948

#### Checklist

- [X] PR description included and stakeholders cc'd
- [X] `npm test` passes
- [X] automated tests are [changed or added](https://testing-dashboard.openfin.co/#/app/dashboard)
- [X] relevant documentation is [changed or added](https://github.com/hadoukenio/js-adapter)
- [X] PR title starts with the JIRA ticket [pull request process](https://github.com/openfin/Internal-Wiki/wiki/Pull-Request-Process)
- [X] PR release notes describe the change in a way relevant to app-developers
- [X] PR has assigned reviewers

#### Release Notes

Notes: 

Adds the following experimental emitted events for BrowserView Methods: 
`created`
`attached`
`shown`
`hidden`
`destroyed`

### CODE REVIEW

File: src/api/browserview/browserview.ts [hunk 1]
```diff
@@ -81,11 +82,11 @@ export class BrowserView extends WebContents<BaseEventMap> {
         const ack = await this.wire.sendAction('get-browser-view-info', {...this.identity});
         return ack.payload.data;
     }
-
     /**
     * Retrieves the window the view is currently attached to.
     * @experimental
-    * @return {Promise.<_Window>}
+    * @return {Promise.<
```
Comment (pbaize) [inline src/api/browserview/browserview.ts]: Invalid JSDoc

File: src/api/events/window.ts [hunk 2]
```diff
@@ -211,8 +192,6 @@ export interface PropagatedWindowEventMapping<Topic = string, Type = string> ext
     'window-performance-report': WindowPerformanceReport<Topic, Type>;
     'window-preload-scripts-state-changed': WindowPreloadScriptsStateChangeEvent<Topic, Type>;
     'window-preload-scripts-state-changing': WindowPreloadScriptsStateChangedEvent<Topic, Type>;
-    'window-resource-load-failed': WindowResourceLoadFailedEvent<Topic, Type>;
```
Comment (pbaize) [inline src/api/events/window.ts]: These should be left in unless we start propagating webcontents events differently

File: tutorials/BrowserView.EventEmitter.md [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+
+### Supported BrowserView event types
+
+* created
+* attached
+* shown
+* hidden
+* destroyed
+
+### Application Events
```
Comment (pbaize) [inline tutorials/BrowserView.EventEmitter.md]: BrowserView

File: src/api/browserview/browserview.ts (hunk 1 above)
Comment (MichaelMCoates) [reply src/api/browserview/browserview.ts]: Good catch, must have happened in the conflict resolution

File: src/api/events/window.ts (hunk 2 above)
Comment (MichaelMCoates) [reply src/api/events/window.ts]: This is from your `hold` commit: 
https://github.com/HadoukenIO/js-adapter/pull/343/commits/9a7d9abc5b2f3d66019e0861c5eb0509ce1287a3#diff-ba47ceacf9a1245b0bb67531547c3d20L204

Comment (pbaize) [reply src/api/events/window.ts]: I messed up then

Comment (pbaize) [reply src/api/events/window.ts]: We might move it into a `propagatedWebcontentsEventMapping` when we add event propagation but for now this should stay.

Comment (MichaelMCoates) [reply src/api/events/window.ts]: Re-added

**Pull request merged**